dojo/tools/bugcrowd_api/api_client.py

```diff
@@ -0,0 +1,104 @@
+        
+        while next != '':
+            if target:
+                response = self.session.get(
+                    url=next,
+                    headers=self.get_headers(),
+                )
+            else:
+                response = self.session.get(
+                    url=next,
+                    headers=self.get_headers(),
+                )
+
+            if response.ok:
+                data = response.json()
+                if len(data['data']) == 0 or data['meta']['total_hits'] == data['meta']['count']:
+                    break
+
+                print('Fetched ' + str(len(data['data'])) + ' submissions')
+                output['data'] = output['data'] + data['data']
+                next='{}{}'.format(self.bugcrowd_api_url,data["links"]["next"])
+            else:
+                next='over'
+                raise Exception("Unable to get asset findings due to {} - {}".format(
+                response.status_code, response.content.decode("utf-8")
+                ))
+        
+        print('Total gathered submissions from Bugcrowd: ' + str(len(output['data'])))
+        
+        return output
+
+
+    def test_connection(self):
+        # Request programs
+        response_programs = self.session.get(
+            url='{}/programs'.format(self.bugcrowd_api_url),
+            headers=self.get_headers(),
+        )
+
+        # Request assets to validate the org token
+        response_subs = self.session.get(
+            url='{}/submissions'.format(self.bugcrowd_api_url),
+            headers=self.get_headers(),
+        )
+
+        if response_programs.ok and response_subs.ok:
+            data = response_programs.json().get('data')
+            progs = list(filter(lambda prog: prog["type"] == "program", data))
+            program_names = ', '.join(list(map(lambda p: p["attributes"]["name"], progs)))
+            return f'You have access to the "{ program_names }" programs'
+        else:
+            raise Exception("Connection failed (error: {} - {})".format(
+                response_assets.status_code, response_assets.content.decode("utf-8")
+            ))
+
+    def test_product_connection(self, api_scan_configuration):
+        submissions = self.get_findings(api_scan_configuration.service_key_1, api_scan_configuration.service_key_2)
+        submission_number = len(submissions['data'])
+        return f'You have access to "{submission_number}" submissions in Bugcrowd in the Program code "{api_scan_configuration.service_key_1}" and Target "{api_scan_configuration.service_key_2}" (leave service key 2 empty to get all submissions in program)'
+
+    def get_headers(self):
+        headers = {
+            'Accept': 'application/vnd.bugcrowd+json',
+            'Authorization': 'Token {}'.format(self.api_token),
+            'User-Agent': 'DefectDojo',
+            'Bugcrowd-Version': '2021-10-28'
+        }
+
+        return headers
```

Comment:
Make it a constant. better

Reply:
If I need the self.api_token I'm not sure how I can make it a constant of the class ?

Reply:
at least make it a constant of the module.
And in the code, just add 'Auth' header.
Something is not good here. I think you should switch to use sessions and set the auth in the session.

Reply:
What difference does it make between using session vs just a constant header ? Is there caching in the client or something ?

Reply:
yes. and the code is clear and structured. auth will be set only one time when building the session.
I can show you. it will also mean fewer lines, fewer bugs, less maintenance... etc

Reply:
Done ! I definitely saw that a lot of clients don't use session yet lol